app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -58,6 +61,8 @@ public static ScreenGrabber getInstance() {
     @TargetApi(Build.VERSION_CODES.LOLLIPOP)
     public void exit() {
         if (projection != null) {
+            virtualDisplay.release();
```

Comment:
I don't think this is needed. Once we lose the reference to the all the existing private variables (by setting them to null below) it's going to get garbage collected (and released and whatever else would be needed). The only existing reference to the Virtual display before this change would be inside the "projection".

But I'm not going to push you to drop this change. No strong opinion on this.

I think this is also the case for the imageReader.close() you added below in another commit.